ed_sensor_integration/src/kinect/ray_tracer.cpp

```diff
@@ -19,16 +19,21 @@ class PointRenderResult : public geo::LaserRangeFinder::RenderResult
 
 public:
 
-    PointRenderResult() : geo::LaserRangeFinder::RenderResult(dummy_ranges_) {}
+    PointRenderResult() : geo::LaserRangeFinder::RenderResult(dummy_ranges_), depth_(0.0), entity_("") {}
 
     void renderPoint(int index, float depth)
     {
         float old_depth = depth_;
-        if (old_depth == 0 || depth < old_depth)
+        if (old_depth == 0.0 || depth < old_depth)
         {
+            ROS_DEBUG_STREAM("Using " << active_entity_ << " as active entity, depth from " << old_depth << " to " << depth);
             depth_ = depth;
             entity_ = active_entity_;
         }
+        else
+        {
+            ROS_DEBUG_STREAM("Not using entity " << active_entity_);
```

Comment:
Not sure about the performance influence of this.

Reply:
Idem